srcpkgs/mold/template

```diff
@@ -0,0 +1,45 @@
+# Template file for 'mold'
+pkgname=mold
+version=0.9.1
+revision=1
```

Comment:
archs="x86_64*"

Reply:
@q66 why? the build for other archs is fixable

Reply:
@q66 please help with this: https://github.com/void-linux/void-packages/pull/31706/checks?check_run_id=2961521405#step:8:336
Why is `chroot-grep` built without `PCRE` support?

Reply:
"Currently, mold is being developed with Linux/x86-64 as the primary target platform. mold can link many user-land programs including large ones such as web browsers for that target. It also has preliminary Linux/i386 support. Supporting other OSes and ISAs are planned after Linux/x86-64 support is complete."

Reply:
that line should be in a comment so people know why archs is there

Reply:
it's right there in the project readme, which @Logarithmus apparently hasn't read

Reply:
> it's right there in the project readme, which @Logarithmus apparently hasn't read

Please, don't make assumptions about the others if you don't know the whole story. I've read the README multiple times, even contributed to it (https://github.com/rui314/mold/pull/63).

This line states that `mold` does support only x86_64 **as a target platform**. The line doesn't say anything about supported **host platform**. So I tried to cross build `mold` for `aarch64` on my laptop and it finised successfully. CI run for `aarch64` worked as well: https://github.com/void-linux/void-packages/pull/31706/checks?check_run_id=2961521454.

After you've posted [this comment](https://github.com/void-linux/void-packages/pull/31706#discussion_r662276527), I asked the author of `mold` if it supports cross linking. The answer was "yes". You can read it [here](https://github.com/rui314/mold/issues/55#issuecomment-872239804). I didn't try it in practice yet though.

Summing up: your suggestion about adding `cross=x86_64*` is wrong.

Reply:
so, you admit you haven't verified it supports cross linking before i posted my comment

CI for cross architectures is pointless when it comes to verifying whether things work, since tests do not run

the linker still defaults to x86_64 mode when run without arguments (rather than explicitly erroring about unsupported architecture), and the usefulness of using an experimental linker to cross-link from some other arch to x86_64 is questionable, so i'd still make it archs=x86_64*

Reply:
@q66
>CI for cross architectures is pointless when it comes to verifying whether things work, since tests do not run

Is it possible to use `qemu` to fix that? Just asking about theoretical possibility, and not the complexity.

>the linker still defaults to x86_64 mode when run without arguments (rather than explicitly erroring about unsupported architecture)

Are talking about `mold` specifically or about linkers in general? If about `mold`, I guess the error message can be added quite easily, since the author of `mold` is a very nice person and accepts contributions without much bureocracy.

>the usefulness of using an experimental linker to cross-link from some other arch to x86_64 is questionable

Please, don't decide for the other Void users, noone forces personally you to use `mold`. Like Arch, Void is a popular distro among Linux users who like experimenting, and having `mold` in a repos for other architectures certainly won't harm anyone. About "experimental": if you didn't know, `mold`'s author is also an original author of `lld`. Also read the release notes for `v0.9.0`: https://github.com/rui314/mold/releases/tag/v0.9#:~:text=We%20bumped%20the%20mold%27s%20version%20from%200.1.1%20to%200.9%20to%20show%20that%20mold%20is%20getting%20ready%20for%20production.

>so i'd still make it archs=x86_64*

You would make it `archs=x86_64*`, but I don't see a reason for this sort of gatekeeping. You sound like if Void is a super-stable Debian-like distro where every packages is supposedly needed to be very thoroughly tested before geting into the repo, but it's simply not true.

Reply:
on the contrary, you  seem to be deciding that things do not need to be well tested before getting into the repo, which is simply not true

Reply:
@q66 I'm all in for well package testing. My point is that IMO many Void packages hasn't been tested at all on all those non x86_64 architectures, especially `armv6`, but they don't have `archs=x86_64*` in their templates.
Correct me if I'm wrong.

Reply:
1) in the past things have been overly lax (until recently there wasn't even CI with unit tests)
2) with most things it's generally safe to assume things work okay, since most things don't have platform specific code, and weren't developed with any specific platform in mind
3) this is a thing that explicitly has platform specific code, and is developed with a single platform in mind for the time being

considering making this work for cross targets involves special compiler setup that complicates the template, and fixing other targets appears to require patching, and even if you do make it build it will not be tested, i don't see any reason to include it in the packages - and i can pretty much guarantee you that there will be zero users trying to link with mold from an arm/ppc/whatever host to x86_64 target (until recently we didn't even ship cross-compilers to x86_64 and nobody complained for years, until i went ahead and added them)

Reply:
@q66
>until recently we didn't even ship cross-compilers to x86_64 and nobody complained for years, until i went ahead and added them

Honestly, did you test all of those cross-compilers before adding them? :)

> making this work for cross targets involves special compiler setup that complicates the template

5 lines of bash, muh much complicated, many code, compared to
```
$ find srcpkgs -name template | xargs wc -l | sort -kh1 
    343 srcpkgs/php/template
    359 srcpkgs/rust/template
    361 srcpkgs/electron10/template
    369 srcpkgs/gcc6/template
    385 srcpkgs/electron12/template
    399 srcpkgs/texlive/template
    423 srcpkgs/llvm12/template
    435 srcpkgs/boost/template
    441 srcpkgs/mesa/template
    661 srcpkgs/libreoffice/template
    723 srcpkgs/gcc/template
   1164 srcpkgs/qt5/template
   1220 srcpkgs/tesseract-ocr/template
```

Reply:
> Honestly, did you test all of those cross-compilers before adding them? :)
> 
of course i did, i cross-bootstrapped the entire system with them

Reply:
@q66 do you have all those boards with `armv6`, `armv7`, etc. CPUs? Or did you use `qemu`?
When I'll resolve all `mold` cross-build issues, can I kindly ask you to run the tests (it should take only a couple of minutes for each arch)?

Reply:
Please set `archs="x86_64*"`

Reply:
@sgn, @q66 OK, you won, I'll set `archs=x86_64*` for now, because right now I don't have any ARM boards to test on, and also now I have no time to dig into emulation of arm using qemu.